core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -474,7 +474,7 @@ class ReplicaManager(val config: KafkaConfig,
         topicPartition ->
                 ProducePartitionStatus(
                   result.info.lastOffset + 1, // required offset
-                  new PartitionResponse(result.error, result.info.firstOffset, result.info.logAppendTime, result.info.logStartOffset)) // response status
+                  new PartitionResponse(result.error, result.info.firstOffset.get, result.info.logAppendTime, result.info.logStartOffset)) // response status
```

Comment:
Don't we need to replace this `get` with `getOrElse(-1)` since we could be getting the `UnknownLogAppendInfo` here? Same thing below.